artemis-server/src/main/java/org/apache/activemq/artemis/spi/core/security/jaas/ReloadableProperties.java

```diff
@@ -95,6 +99,21 @@ public synchronized ReloadableProperties obtained() {
       return invertedValueProps;
    }
 
+   public synchronized Map<String, Pattern> regexpPropertiesMap() {
```

Comment:
You could use a [Supplier::memoize](https://google.github.io/guava/releases/19.0/api/docs/com/google/common/base/Suppliers.html#memoize(com.google.common.base.Supplier) to allow a thread-safe lazy initialization without having that method synchronized even when you just need to get `regexpProps`